### DISCUSSION TITLE
RNode Windows issues

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#499] [General]
Be advised: The way Windows uses com ports (at least the drivers I have) causes RNode problems. We're working on a fix for configuration issues, but some notes:

Windows resets RNodes (and all ESP32s and ATMEGAs) on serial connect and halts them on disconnect. This does not cause problems for RNSD as far as I know, since it's actively connected via serial, but may cause unexpected failures on Bluetooth. If you plug in an RNode to a Windows system, you can use it to power a Bluetooth enabled device, but if you connect to it via serial, then disconnect (for example: connecting to RNS or running RNodeConf) then it will halt on disconnect and no longer function over Bluetooth. While the display will remain on, the MCU is shut down, and the reset button won't function.

If you need to power a Bluetooth RNode from a Windows machine, use a power-only cable or be sure to never connect over serial. If the RNode doesn't appear to respond and the reset button doesn't do anything, unplug it and plug it back in, since it's likely Windows shut it down.

I'm uncertain there's an acceptable solution. Any work-arounds I know of involve hardware modification (which will prevent using the bootloader) or messing with Windows drivers, which even I'm not about to do casually.

**Comment by cobraPA** [comment]
"Windows resets RNodes (and all ESP32s and ATMEGAs) on serial connect and halts them on disconnect."

I'm kind of surprised at this statement...  I don't think I have seen this behavior here using python serial connect on Windows 11.  I'm doing most of my RNode development on that system and the RNode auto install certainly doesn't seem to 'reset the ESP32' boards that I have tried.

What are you using to serial connect to your boards?

**Reply by faragher** [reply]
Windows 10 Pro (upgrade in place over 7) connecting using RNodeConf in the terminal.

USB Serial Device, driver version 10.0.19041.906 (21JUN2006).
